src/components/standalone/firewall/nat/EditNatHelperDrawer.vue

```diff
@@ -0,0 +1,265 @@
+import {
+  NeInlineNotification,
+  NeSideDrawer,
+  NeButton,
+  NeTextInput,
+  getAxiosErrorMessage
+} from '@nethesis/vue-components'
+import { NeToggle } from '@nethesis/vue-components'
+import type { NatHelper } from '@/stores/standalone/firewall'
+import { upperFirst } from 'lodash-es'
+import { useNotificationsStore } from '@/stores/notifications'
+import { useRouter } from 'vue-router'
+import { getStandaloneRoutePrefix } from '@/lib/router'
+
+type Param = {
+  name: string
+  value: string
+}
+
+const props = withDefaults(
+  defineProps<{
+    isShown: boolean
+    natHelper?: NatHelper
+  }>(),
+  { isShown: false }
+)
+
+const emit = defineEmits(['close', 'reloadData'])
+
+const { t } = useI18n()
+const notificationsStore = useNotificationsStore()
+const router = useRouter()
+
+const module = ref('')
+const enabled = ref(true)
+const params = ref<Param[]>([])
+const errorBag = ref(new MessageBag())
+
+const loading = ref({
+  editNatHelper: false
+})
+
+const error = ref({
+  editNatHelper: '',
+  editNatHelperDetails: ''
+})
+
+watch(
+  () => props.isShown,
+  () => {
+    if (props.isShown) {
+      clearErrors()
+
+      if (props.natHelper) {
+        module.value = props.natHelper.name
+        enabled.value = props.natHelper.enabled
+        const moduleParams = []
+
+        if (props.natHelper.params) {
+          for (let [paramName, paramValue] of Object.entries(props.natHelper.params)) {
+            moduleParams.push({
+              name: paramName,
+              value: paramValue
+            })
+          }
+          params.value = moduleParams
+        }
+      }
+    }
+  }
+)
+
+function clearErrors() {
+  errorBag.value.clear()
+  error.value.editNatHelper = ''
+  error.value.editNatHelperDetails = ''
+}
+
+function runValidators(validators: validationOutput[], label: string): boolean {
+  for (let validator of validators) {
+    if (!validator.valid) {
+      errorBag.value.set(label, [validator.errMessage as string])
+    }
+  }
+  return validators.every((validator) => validator.valid)
+}
+
+function validate() {
+  clearErrors()
+
+  if (!enabled.value) {
+    return true
+  }
+
+  // all parameters are required
+
+  const paramValidators: [validationOutput[], string][] = []
+
+  for (let param of params.value) {
+    paramValidators.push([[validateRequired(param.value)], param.name])
+  }
+
+  return paramValidators
+    .map(([validator, label]) => runValidators(validator, label))
+    .every((result) => result)
+}
+
+async function editNatHelper() {
+  const isValidationOk = validate()
+  if (!isValidationOk) {
+    return
+  }
+
+  loading.value.editNatHelper = true
+  const natHelperName = props.natHelper?.name
+
+  const payload = {
+    name: natHelperName,
+    enabled: enabled.value,
+    // convert params array to object
+    params: params.value.reduce((acc, param) => {
+      acc[param.name] = param.value
+      return acc
+    }, {} as Record<string, string>)
```

Comment:
Is this the proper way to use reduce? Isn't a `map` method better?

Reply:
Definitely not liking it, but okay (confirming that AI won't be able to steal our jobs)

Reply:
You're right, I wasn't sure either at first :joy: However we have an explanatory human comment above that line: `// convert params array to object` :innocent: